pkg/app/pipedv1/plugin/kubernetes/livestate/plugin.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package livestate
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	kubeconfig "github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/config"
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/provider"
+	"github.com/pipe-cd/pipecd/pkg/app/pipedv1/plugin/kubernetes/toolregistry"
+	config "github.com/pipe-cd/pipecd/pkg/configv1"
+	"github.com/pipe-cd/pipecd/pkg/plugin/sdk"
+)
+
+type Plugin struct{}
+
+// GetLivestate implements sdk.LivestatePlugin.
+func (p Plugin) GetLivestate(ctx context.Context, _ sdk.ConfigNone, deployTargets []*sdk.DeployTarget[kubeconfig.KubernetesDeployTargetConfig], input *sdk.GetLivestateInput) (*sdk.GetLivestateResponse, error) {
+	if len(deployTargets) != 1 {
+		return nil, fmt.Errorf("only 1 deploy target is allowed but got %d", len(deployTargets))
+	}
+
+	deployTarget := deployTargets[0]
+	deployTargetConfig := deployTarget.Config
+
+	cfg, err := config.DecodeYAML[*kubeconfig.KubernetesApplicationSpec](input.Request.DeploymentSource.ApplicationConfig)
+	if err != nil {
+		input.Logger.Error("Failed to decode the application spec", zap.Error(err))
+		return nil, err
+	}
+
+	// TODO: find the way to hold the tool registry and loader in the plugin.
+	// Currently, we create them every time the stage is executed beucause we can't pass input.Client.toolRegistry to the plugin when starting the plugin.
+	toolRegistry := toolregistry.NewRegistry(input.Client.ToolRegistry())
+
+	// Get the kubectl tool path.
+	kubectlPath, err := toolRegistry.Kubectl(ctx, cmp.Or(cfg.Spec.Input.KubectlVersion, deployTargetConfig.KubectlVersion))
+	if err != nil {
+		input.Logger.Error("Failed to get kubectl tool", zap.Error(err))
+		return nil, err
+	}
+
+	// Create the kubectl wrapper for the target cluster.
+	kubectl := provider.NewKubectl(kubectlPath)
+
+	namespacedLiveResources, clusterScopedLiveResources, err := provider.GetLiveResources(ctx, kubectl, deployTargetConfig.KubeConfigPath, input.Request.ApplicationID)
+	if err != nil {
+		input.Logger.Error("Failed to get live resources", zap.Error(err))
+		return nil, err
+	}
```

Comment:
I noticed that we need to consider the piped spec `spec.platformProviders[].appStateInformer` to select the managing resources. 
https://pipecd.dev/docs-v0.50.x/user-guide/managing-piped/configuration-reference/#kubernetesappstateinformer

But we can do it in another PR. Can you add comment for it?

Reply:
Thank you, I missed it.
I added a comment on this commit. https://github.com/pipe-cd/pipecd/pull/5645/commits/2fc776b2125a535f605d886ebda353a4aa809ff5